x/bech32ics20/keeper/msg_server.go

```diff
@@ -0,0 +1,149 @@
+package keeper
+
+import (
+	"context"
+
+	"github.com/armon/go-metrics"
+
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/bech32"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/bank/types"
+
+	ibctransfertypes "github.com/cosmos/cosmos-sdk/x/ibc/applications/transfer/types"
+	clienttypes "github.com/cosmos/cosmos-sdk/x/ibc/core/02-client/types"
+)
+
+type msgServer struct {
+	Keeper
+}
+
+// NewMsgServerImpl returns an implementation of the bank MsgServer interface
+// for the provided Keeper.
+func NewMsgServerImpl(keeper Keeper) types.MsgServer {
+	return &msgServer{Keeper: keeper}
+}
+
+var _ types.MsgServer = msgServer{}
+
+func (k msgServer) Send(goCtx context.Context, msg *types.MsgSend) (*types.MsgSendResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	if err := k.bk.SendEnabledCoins(ctx, msg.Amount...); err != nil {
+		return nil, err
+	}
+
+	from, err := sdk.AccAddressFromBech32(msg.FromAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	prefix, _, err := bech32.DecodeAndConvert(msg.FromAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	nativePrefix, err := k.hrpToChannelMapper.GetNativeHrp(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if prefix == nativePrefix {
+
```

Comment:
Have this just use the bankKeeper's function (i.e. ` k.bk.Send`)

Reply:
Is it secure to use other module's MsgServer on a module? there's no function called k.bk.Send but only msgServer.Send is available.

Reply:
Yeah, its safe to use the other modules MsgServer, though we should probably be using the other modules keeper directly, since thats what most of the SDK does.